ambry-tools/src/test/java/com.github.ambry/store/BlobIdTransformerTest.java

```diff
@@ -0,0 +1,388 @@
+import com.github.ambry.messageformat.MessageFormatException;
+import com.github.ambry.messageformat.MessageFormatInputStream;
+import com.github.ambry.messageformat.PutMessageFormatBlobV1InputStream;
+import com.github.ambry.messageformat.PutMessageFormatInputStream;
+import com.github.ambry.utils.Utils;
+
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests the BlobIdTransformer
+ */
+public class BlobIdTransformerTest {
+
+  private final ClusterMap clusterMap = new MockReadingClusterMap();
+
+  private final BlobIdFactory blobIdFactory = new BlobIdFactory(clusterMap);
+
+  private final BlobIdTransformer transformer;
+
+  private final List<BlobIdStringPair> blobIdStringPairList;
+
+  private final MockStoreKeyConverter mockStoreKeyConverter;
+
+  private static final int BLOB_STREAM_SIZE = 128;
+  private static final int BLOB_ENCRYPTION_KEY_SIZE = 32;
+  private static final int USER_META_DATA_SIZE = 64;
+  public static final BlobIdStringPair BLOB_ID_PAIR_VERSION_1_CONVERTED = new BlobIdStringPair("AAEAAQAAAAAAAAAhAAAAJDkwNTUwOTJhLTc3ZTAtNDI4NC1iY2IxLTc2MDZlYTAzNWM4OQ", "AAMB_wE5AAIAAQAAAAAAAAAhAAAAJDkwNTUwOTJhLTc3ZTAtNDI4NC1iY2IxLTc2MDZlYTAzNWM4OQ");
+  public static final BlobIdStringPair BLOB_ID_PAIR_VERSION_2_CONVERTED = new BlobIdStringPair("AAIAAQB8AAIAAQAAAAAAAAAbAAAAJDRiYTE0YzFkLTFjNmUtNDYyNC04ZDcyLTU3ZDQzZjgzOWM4OQ", "AAMBAQB8AAIAAQAAAAAAAAAbAAAAJDRiYTE0YzFkLTFjNmUtNDYyNC04ZDcyLTU3ZDQzZjgzOWM4OQ");
+  public static final BlobIdStringPair BLOB_ID_PAIR_VERSION_3_CONVERTED = new BlobIdStringPair("AAMAAgCgAAMAAQAAAAAAAACEAAAAJDYwMmQ0ZGQxLTQ5NDUtNDg0YS05MmQwLTI5YjVkM2ZlOWM4OQ", "AAMBAgCgAAIAAQAAAAAAAACEAAAAJDYwMmQ0ZGQxLTQ5NDUtNDg0YS05MmQwLTI5YjVkM2ZlOWM4OQ");
+  public static final String VERSION_1_UNCONVERTED = "AAEAAQAAAAAAAABZAAAAJGYwMjRiYzIyLTA4NDMtNGNjMC1iMzNiLTUyOGZmZTA4NWM4OQ";
+  public static final String VERSION_3_UNCONVERTED = "AAMAAAAAAAAAAAAAAAAAAAAAAAAAJDUyYTk2OWIyLTA3YWMtNDBhMC05ZmY2LTUxY2ZkZjY4NWM4OQ";
+
+  private static final Class[] VALID_MESSAGE_FORMAT_INPUT_STREAM_IMPLS =
+      new Class[]{PutMessageFormatInputStream.class, PutMessageFormatBlobV1InputStream.class};
+
+  /**
+   * Sets up common components
+   * @throws IOException
+   */
+  public BlobIdTransformerTest() throws IOException {
+    BlobIdStringPair[] blobIdStringPairs = new BlobIdStringPair[]{BLOB_ID_PAIR_VERSION_1_CONVERTED, BLOB_ID_PAIR_VERSION_2_CONVERTED, BLOB_ID_PAIR_VERSION_3_CONVERTED};
+    mockStoreKeyConverter = createAndSetupMockStoreKeyConverter(blobIdStringPairs);
+    transformer = new BlobIdTransformer(mockStoreKeyConverter, blobIdFactory, null);
```

Comment:
maybe have one test with the cache code exercised?

Reply:
parameterized test to operate once with a storekeyconverter and once with a cache.

Reply:
actually went ahead and removed cache from ctor.  If one wants to use a cache they can just create one with a StoreKeyConverter interface.